dspace-api/src/test/java/org/dspace/app/requestitem/RequestItemTest.java

```diff
@@ -0,0 +1,364 @@
+    }
+
+    @Test(expected = AuthorizeException.class)
+    public void testAuthorizeWithMismatchedToken() throws Exception {
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .withAcceptRequest(true)
+                .withAccessToken("test-token")
+                .withDecisionDate(new Date(System.currentTimeMillis() - 86400000)) // Yesterday
+                .withAccessPeriod(0) // forever
+                .build();
+
+        // The access token should NOT valid so we expect to catch an AuthorizeException
+        // when we call RequestItemService.authorizeByAccessToken
+        requestItemService.authorizeAccessByAccessToken(context, request, bitstream, "invalid-token-123");
+    }
+    @Test(expected = AuthorizeException.class)
+    public void testAuthorizeWithMismatchedBitstream() throws Exception {
+        // Create request for one bitstream
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .withAcceptRequest(true)
+                .withAccessToken("test-token")
+                .build();
+
+        // Create different bitstream
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream otherBitstream = BitstreamBuilder
+                .createBitstream(context, item, is)
+                .withName("OtherBitstream")
+                .build();
+
+        // Try to authorize access to different bitstream
+        requestItemService.authorizeAccessByAccessToken(context, request, otherBitstream, request.getAccess_token());
+    }
+
+    @Test(expected = AuthorizeException.class)
+    public void testAuthorizeWithAllFilesDisabled() throws Exception {
+        // Create request for specific bitstream
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .withAcceptRequest(true)
+                .withAccessToken("test-token")
+                .withAllFiles(false)
+                .build();
+
+        // Create different bitstream
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream otherBitstream = BitstreamBuilder
+                .createBitstream(context, item, is)
+                .withName("OtherBitstream")
+                .build();
+
+        // Try to access different bitstream when allfiles=false
+        requestItemService.authorizeAccessByAccessToken(context, request, otherBitstream, request.getAccess_token());
+    }
+
+    @Test
+    public void testGrantRequestWithAccessPeriod() throws Exception {
+        RequestItem request = RequestItemBuilder
+                .createRequestItem(context, item, bitstream)
+                .build();
+
+        request.setAccept_request(true);
+        request.setDecision_date(new Date(System.currentTimeMillis()));
+        request.setAccess_period(7); // 7 day access
```

Comment:
This is a 7 **second** access period, isn't it?  The comment is just wrong here, unless you meant to multiply `7 * 86400`